test/masq.test.js

```diff
@@ -555,3 +578,50 @@ test('should set a watcher', async (done) => {
   masq.watch('/hello', onChange)
   await masq.put(key, value)
 })
+
+test('should be able to get a notif on change in masq-app with a watcher on masq-lib', async (done) => {
+  expect.assertions(2)
+  let resolvePr1
+  const pr1 = new Promise((resolve) => { resolvePr1 = resolve })
+  let resolvePr2
+  const pr2 = new Promise((resolve) => { resolvePr2 = resolve })
+
+  const onChangeMasqLib = () => {
+    expect(true).toBe(true)
+    resolvePr1()
+  }
+  const onChangeMockMasqApp = () => {
+    expect(true).toBe(true)
+    resolvePr2()
+  }
+
+  await logInWithMasqAppMock()
+  const key = '/hello'
+  const value = { data: 'world' }
+  mockMasqApp.watch(masq.userId, '/hello', onChangeMockMasqApp)
+  masq.watch('/hello', onChangeMasqLib)
+  await mockMasqApp.put(masq.userId, key, value)
+
+  await Promise.all([pr1, pr2])
+  done()
+})
+
+test('put/get should put an item and get in Mock Masq App', async () => {
+  expect.assertions(1)
+  await logInWithMasqAppMock()
+
+  let resolvePr
+  const pr = new Promise((resolve) => { resolvePr = resolve })
+  const onChangeMockMasqApp = () => {
```

Comment:
I see no point to do this. Just call `done()` after `mockMasqApp.watch(masq.userId, '/hello', () => done())`, and check the values returned by watch in the callback (before calling done)

Reply:
I would rather not use async and done at the same time as we would not get an error if done is called before an awaited promise is resolved.

Reply:
It this is really needed, you can just write:
`onChangeMockMasqApp = () => Promise.resolve()

Reply:
Promise.resolve() returns a Promise, I don't think we can set a watcher and then await the promise ?